### PR TITLE
Some small fixes: compilation, null ptr derefence and const qualifiers

### PR DESCRIPTION
**Pull request by uburuntu**

### CODE REVIEW

**Comment by uburuntu** [comment]
I signed it! :0

**Comment by uburuntu** [comment]
Man against machine. Part 2.

File: test/cctest/cctest.h [hunk 1]
```diff
@@ -74,9 +74,11 @@ static inline void CheckEqualsHelper(const char* file, int line,
                                      const char* expected,
                                      const char* value_source,
                                      const char* value) {
-  if ((expected == NULL && value != NULL) ||
-      (expected != NULL && value == NULL) ||
-      (expected != NULL && value != NULL && strcmp(expected, value) != 0)) {
+  if (expected == NULL || value == NULL) {
+      abort();
+    }
```
Comment (floitschG) [inline test/cctest/cctest.h]: The indentation looks off to me.

This is not doing the same thing: if the expected and the value should be NULL you would now abort.

**Comment by floitsch** [comment]
Thanks!

**Comment by uburuntu** [comment]
😉

**Pull request merged**